### PR TITLE
Obsolete unused types, LibraryDependencyType and related

### LINKED ISSUE #11015: Obsolete unused types: LibraryDependencyType and related
LibraryDependencyType  is not used in any restore code. It's a dnx remnant that's not helpful anymore.

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11015

Regression? Last working version:

## Description

LibraryDependencyType is not used in any restore code. It's a dnx remnant that's not helpful anymore.

I'd love to remove this type if others agree, https://github.com/search?q=LibraryDependencyType&type=code doesn't reveal anything of note, mostly clones and copies of nuget & old dnx code.

tldr; Obsoleting, but would love to remove if folks agree with it.

Update: Everything except these 2 usages did not really suggest we'd need a deeper investigation.

https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextBuilderTests.cs
https://github.com/dotnet/uwp-setup/blob/main/src/test/Microsoft.Extensions.DependencyModel.Tests/DependencyContextBuilderTests.cs
Looking at those 2, they are using a type that's not even NuGet :D, it's https://www.nuget.org/packages/Microsoft.DotNet.ProjectModel/. 

So I think it should be safe to remove.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by nkolev92** [comment]
🔔 @NuGet/nuget-client

File: src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyType.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 namespace NuGet.LibraryModel
 {
+    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of a original project.json attempt. This type will go away soon. ")]
```
Comment (zkat) [inline src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyType.cs]:
```suggestion
    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of an original project.json attempt. This type will go away soon. ")]
```

File: src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTypeFlag.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 namespace NuGet.LibraryModel
 {
+    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of a original project.json attempt. This type will go away soon.")]
```
Comment (zkat) [inline src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTypeFlag.cs]:
```suggestion
    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of an original project.json attempt. This type will go away soon.")]
```

File: src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTypeKeyword.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 namespace NuGet.LibraryModel
 {
+    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of a original project.json attempt. This type will go away soon.")]
```
Comment (zkat) [inline src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTypeKeyword.cs]:
```suggestion
    [Obsolete("This type is not used in the current implementation of restore. It is a remnant of an original project.json attempt. This type will go away soon.")]
```

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs [hunk 4]
```diff
@@ -159,15 +159,14 @@ public void PackageSpecReader_FrameworkAssemblyEmptyVersion()
             Assert.Equal(VersionRange.All, range);
         }
 
-        [Fact]
+        [Fact] //TODO NK
```
Comment (zkat) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs]: Dangling TODO. Is this already done?

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs [hunk 5]
```diff
@@ -1298,7 +1293,7 @@ public void GetPackageSpec_WhenDependenciesDependencyAutoreferencedValueIsBool_R
         [InlineData("exclude")]
         [InlineData("include")]
         [InlineData("suppressParent")]
-        [InlineData("type")]
+        //[InlineData("type")]
```
Comment (zkat) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs]:
```suggestion
```

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs [hunk 6]
```diff
@@ -1793,7 +1788,7 @@ public void GetPackageSpec_WhenFrameworksDependenciesDependencyAutoreferencedVal
...
```
Comment (zkat) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs]:
```suggestion
```

**Review by zkat — APPROVED** [review]
just a couple of things but this lgtm

**Review by zivkan — APPROVED** [review]

**Comment by nkolev92** [comment]
@zivkan 
Obsolete only or remove?
I don't see any usages in GH search which isn't shocking given that it's a type that has no impact.

**Comment by zivkan** [comment]
> Obsolete only or remove?

You seem confident enough that nobody is using these APIs, and we're at a major version bump, even if we don't follow SemVer strictly, so it seems a good time to remove it. ✂️

**Pull request merged**